README.md

```diff
@@ -2,79 +2,126 @@
+beacon value. Since you are using the localhost network and mocked
+`RrpBeaconServer`, there is no need whitelist the reader before running the
+`read-beacon` script.
+
+You also do not need to edit the `.env` file, since hardhat will connect to the
+local chain automatically (using the default configuration). It will also
+provide funded accounts to run the scripts below.
+
 ```sh
-npx hardhat run scripts/deploy.js --network localhost
-npx hardhat run scripts/read-beacon.js --network localhost
+npm run deploy --network localhost
+npm run read-beacon --network localhost
+# or
+yarn deploy --network localhost
+yarn read-beacon --network localhost
 ```
 
 ### Remote networks
 
-This will require that you set some parameters in an .env file. You could rename the [.env.example](./.env.example) file in the root of this repo and replace the placeholders with valid values.
+This will require that you set some parameters in an `.env` file. You could copy
+the [.env.example](./.env.example) file in the root of this repo and replace the
+placeholders with valid values.
 
-For instance if you wanted to deploy the contract on [Polygon Mumbai Testnet](https://docs.polygon.technology/docs/develop/network-details/network/) you will need to set `NETWORK=polygon-mumbai` and `PROVIDER_URL=https://rpc-mumbai.matic.today`. Keep in mind that you'll need to also set the `MNEMONIC` of an account that needs to have enough funds.
+For instance if you wanted to deploy the contract on
+[Polygon Mumbai Testnet](https://docs.polygon.technology/docs/develop/network-details/network/)
+you will need to set `NETWORK=polygon-mumbai` and
+`PROVIDER_URL=https://rpc-mumbai.matic.today`. Keep in mind that you'll need to
+also set the `MNEMONIC` of an account that needs to have enough funds.
 
 #### Test
 
+The test command will run the tests defined in the `test` directory.
+
 ```sh
-npx hardhat test test/BeaconReaderExample.test.js --network polygon-mumbai
+npm run test --network polygon-mumbai
+# or
+yarn test --network polygon-mumbai
 ```
 
 #### Script
 
+Use these scripts to deploy smart contracts to the target chain and to read the
+beacon value. Since you are using a remote network you also need to whitelist
+the beacon reader before attempting to call `read-beacon` script.
+
+**WARNING:** Please note that whatever value you set for `NETWORK` in the `.env`
+file must match a file name within the `@api3/services` repository. This is
+because the script does not deploy the mocked `RrpBeaconServer`, but instead
+connects to an existing `RrpBeaconServer` provided by API3.
+
 ```sh
-npx hardhat run scripts/deploy.js --network polygon-mumbai
-npx hardhat run scripts/read-beacon.js --network polygon-mumbai
+npm run deploy --network polygon-mumbai
+npm run whitelist-reader --network polygon-mumbai
+npm run read-beacon --network polygon-mumbai
+# or
+yarn deploy --network polygon-mumbai
+yarn whitelist-reader --network polygon-mumbai
```

Comment:
These 3 steps (`yarn`) were successful for me on `rinkeby`, not on `polygon-mumbai`:

The `deploy` step worked, but `whitelist-reader` failed with the below error. 
May be the naming issue: `polygonMumbai` within https://github.com/api3dao/utility-contracts/tree/main/SelfServeRrpBeaconServerWhitelister/deployments

```sh
> yarn whitelist-reader --network polygon-mumbai
yarn run v1.22.11
$ hardhat run scripts/whitelist-reader.js --network polygon-mumbai
Error: No whitelister contract found for chain: polygon-mumbai
    at /home/derek/api3/beacon-reader-example/node_modules/@api3/services/src/index.ts:75:48
    at step (/home/derek/api3/beacon-reader-example/node_modules/@api3/services/dist/index.js:33:23)
    at Object.next (/home/derek/api3/beacon-reader-example/node_modules/@api3/services/dist/index.js:14:53)
    at /home/derek/api3/beacon-reader-example/node_modules/@api3/services/dist/index.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/home/derek/api3/beacon-reader-example/node_modules/@api3/services/dist/index.js:4:12)
    at whitelistBeaconReader (/home/derek/api3/beacon-reader-example/node_modules/@api3/services/dist/index.js:70:12)
    at main (/home/derek/api3/beacon-reader-example/scripts/whitelist-reader.js:34:24)
    at Object.<anonymous> (/home/derek/api3/beacon-reader-example/scripts/whitelist-reader.js:54:1)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
Yeah, I thought so too. Will update this.

Reply:
Should be fixed by https://github.com/api3dao/utility-contracts/pull/2